src/urllib3/response.py

```diff
@@ -291,8 +291,9 @@ def get_all(self) -> bytes:
         if len(buffer) == 1:
             result = buffer.pop()
         else:
-            result = b"".join(buffer)
-            buffer.clear()
+            ret = io.BytesIO()
+            ret.writelines(buffer.popleft() for _ in range(len(buffer)))
+            result = ret.getvalue()
```

Comment:
Were you able to measure the memory gain? I would expect `str.join` to do exactly that under the hood. If it made a difference, the surely we could lower some of the `pytest.mark.limit_memory` decorators?

Reply:
With `b"".join(buffer)` there was a moment when `result` took memory for all the data, but `ret` has not been cleared yet and it took the same amount of memory additionally. So `test_memory_usage` reported around 20 MB instead of around 12 MB (5 * 2 MiB chunks + 2 MiB for overhead from the last chunk.)